service_discovery/dto/service_proposal.go

```diff
@@ -67,6 +73,55 @@ func unserializePaymentMethod(paymentMethod string, message *json.RawMessage) (P
 	return nil, errors.New("Payment method unserializer '" + paymentMethod + "' doesn't exist")
 }
 
+/**
+ * Contact unserializer registry logic
+ */
+type ContactDefinitionUnserializer func(*json.RawMessage) (ContactDefinition, error)
+
+// service payment method unserializer registry
+var contactDefinitionMap map[string]ContactDefinitionUnserializer = make(map[string]ContactDefinitionUnserializer, 0)
+
+func RegisterContactDefinitionUnserializer(paymentMethod string, unserializer func(*json.RawMessage) (ContactDefinition, error)) {
+	contactDefinitionMap[paymentMethod] = unserializer
+}
+
+func unserializeContactDefinition(message *json.RawMessage) (contactList []Contact, err error) {
+	// get an array of raw definitions
+	var contacts []struct {
+		Type       string           `json:"type"`
+		Definition *json.RawMessage `json:"definition"`
+	}
+	if err = json.Unmarshal([]byte(*message), &contacts); err != nil {
+		return
+	}
+
+	length := len(contacts)
+	if length == 0 {
```

Comment:
I dont think special case for 0 case is needed. Empty list will traverse 0 times